[PhpUnitBridge] Broken with PHPUnit 9.5

**Symfony version(s) affected**: v5.2.0

**Description**  

PhpUnitBridge's CoverageListenerTrait is incompatible with PHPUnit 9.5, probably due to [missing/inaccessible `getAnnotations()` method](https://github.com/symfony/phpunit-bridge/blob/92a76ca5e64effd41ce111b8f476144dfa29f1f0/Legacy/CoverageListenerTrait.php#L45).

**How to reproduce**  

Example of [a build with the error](https://github.com/FriendsOfPHP/PHP-CS-Fixer/runs/1503268631#step:9:17):
```
 PHPUnit 9.5.0 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.13
Configuration: /home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer/phpunit.xml.dist
Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

Testing 
Call to undefined method PhpCsFixer\Tests\AbstractFixerTest::getAnnotations()
Error: Process completed with exit code 2.
```

[And another.](https://github.com/infection/infection/pull/1452/checks?check_run_id=1503353661#step:7:13)

**Workaround**  

Use PHPUnit 9.4.

--- a/phpunit
+++ b/phpunit
@@ -13,7 +13,7 @@ if (!getenv('SYMFONY_PHPUNIT_VERSION')) {
     } elseif (\PHP_VERSION_ID < 70300) {
         putenv('SYMFONY_PHPUNIT_VERSION=8.5');
     } else {
-        putenv('SYMFONY_PHPUNIT_VERSION=9.4');
+        putenv('SYMFONY_PHPUNIT_VERSION=9.5');
     }
 }
 if (!getenv('SYMFONY_PATCH_TYPE_DECLARATIONS')) {

--- a/src/Symfony/Bridge/PhpUnit/Legacy/CoverageListenerTrait.php
+++ b/src/Symfony/Bridge/PhpUnit/Legacy/CoverageListenerTrait.php
@@ -42,7 +42,7 @@ class CoverageListenerTrait
             return;
         }
 
-        $annotations = $test->getAnnotations();
+        $annotations = Test::parseTestMethodAnnotations(\get_class($test), $test->getName(false));
 
         $ignoredAnnotations = ['covers', 'coversDefaultClass', 'coversNothing'];
 

--- a/src/Symfony/Component/Security/phpunit.xml.dist
+++ b/src/Symfony/Component/Security/phpunit.xml.dist
@@ -14,7 +14,6 @@
 
     <testsuites>
         <testsuite name="Symfony Security Component Test Suite">
-            <directory>./Tests/</directory>
             <directory>./*/Tests/</directory>
         </testsuite>
     </testsuites>
@@ -23,8 +22,6 @@
         <whitelist>
             <directory>./</directory>
             <exclude>
-                <directory>./Resources</directory>
-                <directory>./Tests</directory>
                 <directory>./vendor</directory>
                 <directory>./*/Resources</directory>
                 <directory>./*/Tests</directory>

